src/test/java/software/amazon/event/kafkaconnector/EventBridgeSinkTaskTest.java

```diff
@@ -4,19 +4,152 @@
  */
 package software.amazon.event.kafkaconnector;
 
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static java.util.stream.Collectors.toList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.reset;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.sink.ErrantRecordReporter;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
 import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
 import software.amazon.awssdk.services.eventbridge.EventBridgeAsyncClient;
+import software.amazon.awssdk.services.eventbridge.model.EventBridgeException;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequest;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsRequestEntry;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsResponse;
+import software.amazon.awssdk.services.eventbridge.model.PutEventsResultEntry;
 
 @ExtendWith(MockitoExtension.class)
 public class EventBridgeSinkTaskTest {
 
+  private final ObjectMapper mapper = new ObjectMapper();
+
   @Mock private EventBridgeAsyncClient eventBridgeAsyncClient;
+  @Mock private ErrantRecordReporter dlq;
+
+  private final EventBridgeSinkConfig config =
+      new EventBridgeSinkConfig(
+          Map.of(
+              "aws.eventbridge.retries.max", 10,
+              "aws.eventbridge.connector.id", "someId",
+              "aws.eventbridge.region", "eu-central-1",
+              "aws.eventbridge.eventbus.arn", "something",
+              "aws.eventbridge.detail.types", "my-first-${topic}"));
+
+  @BeforeEach
+  void resetMocks() {
+    reset(eventBridgeAsyncClient, dlq);
+  }
+
+  @Test
+  @DisplayName("should send records successfully")
+  public void sendSuccessfully() {
```

Comment:
question: this test implementation seems to assume single request per put, i.e., we need to change it when we support different batching strategies. Should we change the test rather now then?

Reply:
It is the current and non changeable strategy. It will be changed in #26.